drivers: reset: nxp: add nxp-mrcc-reset for MCXA

peterwangsz opened the pull request:
1. the MCXA family use MRCC to reset peripheral
2. MRCC register, 0: hold in reset, 1: release from reset
3. usage as blow
syscon: syscon@xxx {
    ...
    reset: reset {
        compatible = "nxp,mrcc-reset";
        #reset-cells = <1>;
    };
};

decsny reviewed (changes requested):
make another driver, you are ifdeffing around basically every line of code in in this one

peterwangsz committed:
drivers: reset: nxp: add nxp_mrcc_reset driver

1. the MCXA family use MRCC to reset peripheral
2. MRCC register, 0: hold in reset, 1: release from reset
3. usage as blow
syscon: syscon@xxx {
    ...
    reset: reset {
        compatible = "nxp,mrcc-reset";
        #reset-cells = <1>;
    };
};

Signed-off-by: Peter Wang <chaoyi.wang@nxp.com>

peterwangsz commented:
hi @decsny ,
The nxp-mrcc-reset driver was created for MCXA, please help to review, thanks.

decsny commented:
@peterwangsz I see you added the driver and binding, but I thought this is supposed to enable for MCXA, doesn't it need added to MCXA DTS ?

peterwangsz committed:
boards: frdm_mcxa15x: enable reset driver

1. update mcxa to use nxp,mrcc-reset reset driver

Signed-off-by: Peter Wang <chaoyi.wang@nxp.com>

peterwangsz commented:
hi @decsny , I updated the DTS for frdm_mcxa153 and frdm_mcxa156 in new commit and will update frdm_mcxa346,frdm_mcxa266 and frdm_mcxa366 after https://github.com/zephyrproject-rtos/zephyr/pull/95490 is merged.
